### PR TITLE
Delete unused Monitor.Wait argument

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/src/classlibnative/bcltype/objectnative.cpp [hunk 1]
```diff
@@ -254,21 +254,19 @@ FCIMPL1(Object*, ObjectNative::AllocateUninitializedClone, Object* pObjUNSAFE)
...
 
     BOOL retVal = FALSE;
     OBJECTREF pThis = (OBJECTREF) pThisUNSAFE;
     HELPER_METHOD_FRAME_BEGIN_RET_1(pThis);
 
-    if (pThis == NULL)
-        COMPlusThrow(kNullReferenceException, W("NullReference_This"));
-
-    if ((Timeout < 0) && (Timeout != INFINITE_TIMEOUT))
-        COMPlusThrowArgumentOutOfRange(W("millisecondsTimeout"), W("ArgumentOutOfRange_NeedNonNegNum"));
```
Comment (jkotas) [inline src/coreclr/src/classlibnative/bcltype/objectnative.cpp:269]: The error message was not correct. I have fixed it while I was on it.

File: src/coreclr/src/System.Private.CoreLib/src/System/Threading/Monitor.cs [hunk 2]
```diff
@@ -163,45 +163,38 @@ public static bool IsEntered(object obj)
...
+            => Wait(obj, millisecondsTimeout);
 
+        // Remoting is not supported, exitContext argument is unused
         [UnsupportedOSPlatform("browser")]
-        public static bool Wait(object obj)
-        {
-            return Wait(obj, Timeout.Infinite, false);
-        }
+        public static bool Wait(object obj, TimeSpan timeout, bool exitContext)
+            => Wait(obj, MillisecondsTimeoutFromTimeSpan(timeout));
```
Comment (marek-safar) [inline src/coreclr/src/System.Private.CoreLib/src/System/Threading/Monitor.cs:197]: Why not to move them to shared partition?

**Review by stephentoub — APPROVED** [review]

File: src/coreclr/src/System.Private.CoreLib/src/System/Threading/Monitor.cs (hunk 2 above)
Comment (jkotas) [reply src/coreclr/src/System.Private.CoreLib/src/System/Threading/Monitor.cs]: Moved the easy ones.

**Review by CoffeeFlux — APPROVED** [review]
Mono changes LGTM, thanks!

**Pull request merged**